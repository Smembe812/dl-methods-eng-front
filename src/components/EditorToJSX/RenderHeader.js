import React from "react";

function RenderHeader({block, options={}}){
    const {className} = options
    let receivedBlock;
    if(typeof block === "string"){
        receivedBlock = JSON.parse(block)
    }
    else if (typeof block === "object"){
        receivedBlock = block
    }
    else{
        return null
    }
    const {type, data: {text, level}} = receivedBlock
    
    if(type === "header"){
        switch(level){
            case 1:
                if(className){
                    return <h1 className={className}>{text}</h1>
                }
                return <h1>{text}</h1>
            case 2:
                if(className){
                    return <h2 className={className}>{text}</h2>
                }
                return <h2>{text}</h2> 
            case 3:
                if(className){
                    return <h3 className={className}>{text}</h3>
                }
                return <h3>{text}</h3>
            case 4:
                if(className){
                    return <h4 className={className}>{text}</h4>
                }
                return <h4>{text}</h4>
            case 5:
                if(className){
                    return <h5 className={className}>{text}</h5>
                }
                return <h5>{text}</h5>
            case 6:
                if(className){
                    return <h6 className={className}>{text}</h6>
                }
                return <h6>{text}</h6>
            default:
                return null
        }
    }
}

export default RenderHeader