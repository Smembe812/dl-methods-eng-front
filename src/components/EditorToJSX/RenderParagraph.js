import React from "react";

function RenderParagraph({block, options={}}){
    const {className} = options
    let receivedBlock;
    if(typeof block === "string"){
        receivedBlock = JSON.parse(block)
    }
    else if (typeof block === "object"){
        receivedBlock = block
    }
    else{
        return null
    }
    const {type, data: {text}} = receivedBlock
    
    console.log("in null game", type)
    if(type === "paragraph"){
        if (className){
            return <p className={className}>{text}</p>
        }

        return <p>{text}</p>
    }
}

export default RenderParagraph