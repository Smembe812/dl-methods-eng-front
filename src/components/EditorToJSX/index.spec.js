import React from 'react'
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import renderer from 'react-test-renderer';

import RenderHeader from './RenderHeader'
import RenderParagraph from './RenderParagraph'

describe('Convert EditorJS data to JSX', () => {
    let container = null;
    beforeEach((done) => {
      // setup a DOM element as a render target
      container = document.createElement("div");
      document.body.appendChild(container);
      done()
    });
    
    afterEach((done) => {
      // cleanup on exiting
      unmountComponentAtNode(container);
      container.remove();
      container = null;
      done()
    });

    describe("render header", () => {
        const headerBlocks = [
            {
                "type" : "header",
                "data" : {
                    "text" : "Editor.js",
                    "level" : 1
                }
            },
            {
                "type" : "header",
                "data" : {
                    "text" : "Editor.js",
                    "level" : 2
                }
            },
            {
                "type" : "header",
                "data" : {
                    "text" : "Editor.js",
                    "level" : 3
                }
            },
            {
                "type" : "header",
                "data" : {
                    "text" : "Editor.js",
                    "level" : 4
                }
            },
            {
                "type" : "header",
                "data" : {
                    "text" : "Editor.js",
                    "level" : 5
                }
            },
            {
                "type" : "header",
                "data" : {
                    "text" : "Editor.js",
                    "level" : 6
                }
            }
        ]
        it("should render header", (done) => {
            act(() => {
                render(<RenderHeader block={headerBlocks[0]}
                    options={{className:"class-name"}}/>, container);
              });
          
              // const linkElement = getByText(/learn react/i);
              
            expect(container.textContent).toBe("Editor.js");
            done()
        })
    
        it("should render h1 with className", (done) => {
            const tree = renderer
                .create(
                <RenderHeader 
                    block={headerBlocks[0]} 
                    options={{className:"class-name"}}/>)
                .toJSON();
            expect(tree).toMatchSnapshot();
            done()
        })

        it("should render h2 with className", (done) => {
            const tree = renderer
                .create(
                <RenderHeader 
                    block={headerBlocks[1]}
                    options={{className:"class-name"}}/>)
                .toJSON();
            expect(tree).toMatchSnapshot();
            done()
        })

        it("should render h3 with className", (done) => {
            const tree = renderer
                .create(<RenderHeader block={headerBlocks[2]}
                    options={{className:"class-name"}}/>)
                .toJSON()
            expect(tree).toMatchSnapshot();
            done()
        })

        it("should render h4 with className", (done) => {
            const tree = renderer
                .create(<RenderHeader block={headerBlocks[3]}
                    options={{className:"class-name"}}/>)
                .toJSON();
            expect(tree).toMatchSnapshot();
            done()
        })

        it("should render h5 with className", (done) => {
            const tree = renderer
                .create(<RenderHeader block={headerBlocks[4]}
                    options={{className:"class-name"}}/>)
                .toJSON();
            expect(tree).toMatchSnapshot();
            done()
        })

        it("should render h6 with className", (done) => {
            const tree = renderer
                .create(<RenderHeader block={headerBlocks[5]}
                    options={{className:"class-name"}}/>)
                .toJSON();
            expect(tree).toMatchSnapshot();
            done()
        })
    })


    describe("render paragraph", () => {

        const paragraphBlock = [
            {
                "type" : "paragraph",
                "data" : {
                    "text" : "Editor.js"
                }
            }
        ]
        it("should render a pragraph with className", (done) => {
            const tree = renderer
                    .create(
                        <RenderParagraph block={paragraphBlock[0]}
                            options={{className:"class-name"}}/>)
                    .toJSON();
                expect(tree).toMatchSnapshot();
                done()
        })
        

    })
    it.todo("should render")
    it.todo("should render img")
    it.todo("should render lists")
    it.todo("should render table")
    it.todo("should render checklist")
    it.todo("should render paragraph")
    it("Should always run" , (done) => {
        expect(1).toBe(1)
        done()
    })
});

