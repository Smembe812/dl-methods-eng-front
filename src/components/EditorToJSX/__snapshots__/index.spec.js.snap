// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Convert EditorJS data to JSX render header should render h1 with className 1`] = `
<h1
  className="class-name"
>
  Editor.js
</h1>
`;

exports[`Convert EditorJS data to JSX render header should render h2 with className 1`] = `
<h2
  className="class-name"
>
  Editor.js
</h2>
`;

exports[`Convert EditorJS data to JSX render header should render h3 with className 1`] = `
<h3
  className="class-name"
>
  Editor.js
</h3>
`;

exports[`Convert EditorJS data to JSX render header should render h4 with className 1`] = `
<h4
  className="class-name"
>
  Editor.js
</h4>
`;

exports[`Convert EditorJS data to JSX render header should render h5 with className 1`] = `
<h5
  className="class-name"
>
  Editor.js
</h5>
`;

exports[`Convert EditorJS data to JSX render header should render h6 with className 1`] = `
<h6
  className="class-name"
>
  Editor.js
</h6>
`;

exports[`Convert EditorJS data to JSX render paragraph should render a pragraph with className 1`] = `
<p
  className="class-name"
>
  Editor.js
</p>
`;
